pkg/v7/chartconfig/desired_test.go

```diff
@@ -0,0 +1,413 @@
+			if err != nil {
+				t.Fatal("expected", nil, "got", err)
+			}
+
+			if len(chartConfigs) != len(tc.expectedChartConfigNames) {
+				t.Fatal("expected", len(tc.expectedChartConfigNames), "got", len(chartConfigs))
+			}
+
+			for _, expectedName := range tc.expectedChartConfigNames {
+				_, err := getChartConfigByName(chartConfigs, expectedName)
+				if IsNotFound(err) {
+					t.Fatalf("expected chart %s not found", expectedName)
+				} else if err != nil {
+					t.Fatalf("expected nil, got %#v", err)
+				}
+			}
+
+			for _, spec := range tc.providerChartSpecs {
+				chartConfig, err := getChartConfigByName(chartConfigs, spec.ChartName)
+				if err != nil {
+					t.Fatalf("expected nil, got %#v", err)
+				}
+
+				if chartConfig.TypeMeta.Kind != chartConfigKind {
+					t.Fatalf("expected kind %#q got %#q", chartConfigKind, chartConfig.TypeMeta.Kind)
+				}
+				if chartConfig.TypeMeta.APIVersion != chartConfigAPIVersion {
+					t.Fatalf("expected api version %#q got %#q", chartConfigAPIVersion, chartConfig.TypeMeta.APIVersion)
+				}
+
+				if spec.ChartName != chartConfig.ObjectMeta.Name {
+					t.Fatalf("expected chart name %#q got %#q", spec.ChartName, chartConfig.ObjectMeta.Name)
+				}
+				if spec.AppName != chartConfig.ObjectMeta.Labels["app"] {
+					t.Fatalf("expected app label %#q got %#q", spec.AppName, chartConfig.ObjectMeta.Labels["app"])
+				}
+
+				if spec.ChannelName != chartConfig.Spec.Chart.Channel {
+					t.Fatalf("expected channel name %#q got %#q", spec.ChannelName, chartConfig.Spec.Chart.Channel)
+				}
+				if spec.ChartName != chartConfig.Spec.Chart.Name {
+					t.Fatalf("expected chart name %#q got %#q", spec.ChartName, chartConfig.Spec.Chart.Name)
+				}
+				if spec.Namespace != chartConfig.Spec.Chart.Namespace {
+					t.Fatalf("expected chart namespace %#q got %#q", spec.Namespace, chartConfig.Spec.Chart.Namespace)
+				}
+				if spec.ReleaseName != chartConfig.Spec.Chart.Release {
+					t.Fatalf("expected release name %#q got %#q", spec.ReleaseName, chartConfig.Spec.Chart.Release)
+				}
+
+				if spec.ConfigMapName != "" {
+					if spec.ConfigMapName != chartConfig.Spec.Chart.ConfigMap.Name {
+						t.Fatalf("expected configmap name %#q got %#q", spec.ConfigMapName, chartConfig.Spec.Chart.ConfigMap.Name)
+					}
+					if chartConfig.Spec.Chart.ConfigMap.ResourceVersion != "12345" {
+						t.Fatalf("expected configmap resource version '12345' got %#q", chartConfig.Spec.Chart.ConfigMap.ResourceVersion)
+					}
+					if spec.Namespace != chartConfig.Spec.Chart.ConfigMap.Namespace {
+						t.Fatalf("expected configmap namespace %#q got %#q", spec.Namespace, chartConfig.Spec.Chart.ConfigMap.Namespace)
+					}
+				}
+
+				if chartConfig.Spec.VersionBundle.Version != chartConfigVersionBundleVersion {
+					t.Fatalf("expected version bundle version %#q got %#q", chartConfigVersionBundleVersion, chartConfig.Spec.VersionBundle.Version)
+				}
+			}
+		})
+	}
+}
+
```

Comment:
Apart from name change from `getConfigMapSpec` to `newConfigMapSpec` this is just copy paste from the current resource.